public/request/game/modify.php

```diff
@@ -17,37 +16,28 @@
 $value = requestInputPost('v');
 
 if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Developer)) {
-    echo "FAILED!";
+    echo json_encode(['success' => false, 'error' => 'Unauthorized']);
     exit;
 }
 
-if ($field === GameAction::UpdateHash) {
-    $name = requestInputPost('n');
-    $labels = requestInputPost('l');
-    if (updateHashDetails($gameID, $value, $name, $labels)) {
-        // Log hash update
-        addArticleComment("Server", ArticleType::GameHash, $gameID, $value . " updated by " . $user . ". Description: \"" . $name . "\". Label: \"" . $labels . "\"");
-        echo "OK";
+switch ($field) {
+    case GameAction::UnlinkHash:
+        $result = removeHash($user, $gameID, $value);
+        break;
+
+    case GameAction::UpdateHash:
+        $name = requestInputPost('n');
+        $labels = requestInputPost('l');
+        $result = updateHashDetails($user, $gameID, $value, $name, $labels);
+        break;
+
+    default:
+        echo json_encode(['success' => false, 'error' => 'Bad request: invalid field (' . $field . ')'], JSON_THROW_ON_ERROR);
```

Comment:
Per https://github.com/RetroAchievements/RAWeb/pull/1017#discussion_r889418682, I've made this return JSON (the managehashes page already had a lot of logic to refresh the grid without reloading the page).

I've moved the ModifyTitle handler to `game/update.php`, which was already handling form posts for things like the developer/publisher and forum topic.